EIPS/eip-8032.md

```diff
@@ -29,59 +31,153 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
+- slot `0x02`: `cursor_accum` (uint256 count of non-zero slots seen for the current account)
+
+At the **end** of each block, the client performs up to `TRANSITION_SLOTS_PER_BLOCK` storage slot iterations, possibly spanning one account or multiple accounts:
+
+1. **Select account**  
+   If `cursor_account_hash == 0x00…00`, set it to the smallest account hash **strictly greater than** `0x00…00` whose `codeHash` is non-empty. Otherwise, continue with `cursor_account_hash`.
+
+2. **Iterate storage slots**  
+   Visit storage slots of `cursor_account_hash` in ascending lexicographic order of `slot hash`, starting **strictly after** `cursor_slot_hash` (or from the beginning if `cursor_slot_hash == 0x00…00`).
+   - Increment `cursor_accum` slot for every slot hash seen
+   - Stop when either:
+     - `TRANSITION_SLOTS_PER_BLOCK` slots have been processed, or
+     - there are no more storage slots for the account.
+
+3. **Finalize account (if exhausted)**  
+   If all storage slots for `cursor_account_hash` have been visited:
+   - Set `storage_count(cursor_account)` to `cursor_accum` in the account object.
+   - Advance to the next account:  
+     `cursor_account_hash = next account hash with non-empty code hash`  
+   - Reset per-account cursors:  
+     `cursor_slot_hash = 0x00…00`, `cursor_accum = 0`.
+
+4. **Persist**  
+   If `TRANSITION_SLOTS_PER_BLOCK` is reached, write `cursor_account_hash`, `cursor_slot_hash`, and `cursor_accum` to the registry in the post-state of the block.
+
+5. **Completion**  
+   The transition is complete when there is no account with non-empty `codeHash` whose account hash is lexicographically greater than the current `cursor_account_hash`.
+
+The account hashes and slot hashes are iterated in ascending lexicographical order.
+
+Transition reference: [EIP7612](https://eips.ethereum.org/EIPS/eip-7612)
+
+### Account update rules
+
+After executing all transactions in a block, clients MUST update `storage_count` for affected accounts as follows:
+
+- Let `old := storage_count(A)` in the block-prestate (treat as `0` if absent).
+- Let `new` be the true count of non-zero slots for `A` in the post-block state.
+- Set `storage_count(A) := new`.
+
+To compute `new` efficiently without scanning all storage every block, clients MUST apply net deltas derived from slots whose values changed relative to the block-prestate:
+
+For each storage slot `k` of account `A` that was written at least once in the block:
 
-where `compute_root_and_max_depth` is a function that writes all storage changes to the tree, and return the maximum depth that a storage slot has been written to during that process.
+$$
+\delta = \begin{cases}
+1 & \text{if } v_0 = 0 \wedge v_1 \ne 0 \\
+-1 & \text{if } v_0 \ne 0 \wedge v_1 = 0 \\
+0 & \text{otherwise}
+\end{cases}
+$$
 
-### Gas cost changes
+During a transition, the transition is advanced first before applying the storage deltas. Update semantics when a transition is occuring:
```

Comment:
Add a note that this happens _before_ applying the updates, but _after_ execution, and so the gas costs of a contract that is sweeped by the iterator in this block, are only applied for the _next_ block